### PR TITLE
fix: add date to job name

### PR DESCRIPTION
**Pull request by BUAA-yang-wb**
### 改动说明
在作业名称中添加日期前缀（YYMMDD 格式），便于识别
### 具体改动
- Jupyter 作业名格式：`jupyter-{username}-{YYMMDD}-{uuid}`
- 单机训练作业名格式：`single-{username}-{YYMMDD}-{uuid}`
- PyTorch 作业名格式：`py-{username}-{YYMMDD}-{uuid}`
- TensorFlow 作业名格式：`tf-{username}-{YYMMDD}-{uuid}`
- Sparse 作业名格式：`sparse-{username}-{YYMMDD}-{uuid}`

### CODE REVIEW

File: backend/pkg/crclient/jobclient.go [hunk 1]
```diff
@@ -145,7 +151,11 @@ func (c *JobControl) createJupyterJobFromTask(task *model.AITask) (jobname strin
 
 	// convert metadata to lower case
 	taskName := strings.ToLower(task.TaskName)
-	jobname = fmt.Sprintf("%s-%d", taskName, task.ID)
+	// generate date in yymmdd format
+	now := time.Now()
+	dateStr := fmt.Sprintf("%02d%02d%02d", now.Year()%twoDigitYearDivisor, int(now.Month()), now.Day())
```
Comment (Jinghao-coding) [inline backend/pkg/crclient/jobclient.go:156]: 这里给Month加了int(), 其他地方呢

File: backend/internal/handler/vcjob/tensorflow.go [hunk 2]
```diff
@@ -79,8 +80,10 @@ func (mgr *VolcanojobMgr) CreateTensorflowJob(c *gin.Context) {
 		return
 	}
 
-	// Ingress base URL
-	baseURL := fmt.Sprintf("%s-%s", token.Username, uuid.New().String()[:6])
+	// Ingress base URL with date (YYMMDD format)
+	now := time.Now()
+	dateStr := fmt.Sprintf("%02d%02d%02d", now.Year()%twoDigitYearDivisor, now.Month(), now.Day())
```
Comment (Jinghao-coding) [inline backend/internal/handler/vcjob/tensorflow.go:85]: 这里

File: backend/internal/handler/vcjob/tensorflow.go [hunk 3]
```diff
@@ -79,8 +80,10 @@ func (mgr *VolcanojobMgr) CreateTensorflowJob(c *gin.Context) {
...
+	baseURL := fmt.Sprintf("%s-%s-%s", token.Username, dateStr, uuid.New().String()[:6])
```
Comment (Jinghao-coding) [inline backend/internal/handler/vcjob/tensorflow.go:86]: 这里uuid是6位 其他地方是5位？

File: backend/internal/handler/vcjob/pytorch.go [hunk 4]
```diff
@@ -43,8 +44,10 @@ func (mgr *VolcanojobMgr) CreatePytorchJob(c *gin.Context) {
 		return
 	}
 
-	// base URL
-	baseURL := fmt.Sprintf("%s-%s", token.Username, uuid.New().String()[:5])
+	// base URL with date (YYMMDD format)
+	now := time.Now()
+	dateStr := fmt.Sprintf("%02d%02d%02d", now.Year()%twoDigitYearDivisor, now.Month(), now.Day())
+	baseURL := fmt.Sprintf("%s-%s-%s", token.Username, dateStr, uuid.New().String()[:5])
```
Comment (Jinghao-coding) [inline backend/internal/handler/vcjob/pytorch.go:50]: 这里

File: backend/internal/handler/vcjob/jupyter.go [hunk 5]
```diff
@@ -79,8 +81,10 @@ func (mgr *VolcanojobMgr) CreateJupyterJob(c *gin.Context) {
 		return
 	}
 
-	// Ingress base URL
-	baseURL := fmt.Sprintf("%s-%s", token.Username, uuid.New().String()[:5])
+	// Ingress base URL with date (YYMMDD format)
+	now := time.Now()
+	dateStr := fmt.Sprintf("%02d%02d%02d", now.Year()%twoDigitYearDivisor, now.Month(), now.Day())
+	baseURL := fmt.Sprintf("%s-%s-%s", token.Username, dateStr, uuid.New().String()[:5])
```
Comment (Jinghao-coding) [inline backend/internal/handler/vcjob/jupyter.go:87]: 这里

File: backend/internal/handler/spjob/spjob.go [hunk 6]
```diff
@@ -35,7 +36,10 @@ func init() {
 	}
 }
 
-const AnnotationKeyTaskName = "crater.raids.io/task-name"
+const (
+	AnnotationKeyTaskName = "crater.raids.io/task-name"
```
Comment (Jinghao-coding) [inline backend/internal/handler/spjob/spjob.go:40]: twoDigitYearDivisor 每个文件中都定义了一下  是不是有更好的解决办法？

Comment (Jinghao-coding) [reply backend/internal/handler/spjob/spjob.go:40]: BaseURL这部分逻辑是不是可以抽象出来 这样不会重复写代码了

File: backend/internal/handler/spjob/spjob.go [hunk 7]
```diff
@@ -127,7 +131,10 @@ func (mgr *SparseJobMgr) Create(c *gin.Context) {
 		return
 	}
 
-	baseURL := fmt.Sprintf("%s-%s", token.Username, uuid.New().String()[:5])
+	// Ingress base URL with date (YYMMDD format)
+	now := time.Now()
+	dateStr := fmt.Sprintf("%02d%02d%02d", now.Year()%twoDigitYearDivisor, now.Month(), now.Day())
+	baseURL := fmt.Sprintf("%s-%s-%s", token.Username, dateStr, uuid.New().String()[:5])
```
Comment (Jinghao-coding) [inline backend/internal/handler/spjob/spjob.go:137]: 这里也是5位

File: backend/internal/handler/vcjob/custom.go [hunk 8]
```diff
@@ -65,8 +66,10 @@ func (mgr *VolcanojobMgr) CreateTrainingJob(c *gin.Context) {
 		return
 	}
 
-	// base URL
-	baseURL := fmt.Sprintf("%s-%s", token.Username, uuid.New().String()[:5])
+	// Ingress base URL with date (YYMMDD format)
+	now := time.Now()
```
Comment (Jinghao-coding) [inline backend/internal/handler/vcjob/custom.go:70]: 这里其他代码怎么写的，要不要统一时区，最好问一下亦龙或者怡丰

File: backend/internal/handler/spjob/spjob.go (hunk 6 above)
Comment (Jinghao-coding) [reply backend/internal/handler/spjob/spjob.go:40]: 最终生成的名称长度有判断吗？会不会超过长度？看看能不能写个单测验证一下

File: backend/internal/handler/vcjob/pytorch.go [hunk 9]
```diff
@@ -43,8 +44,10 @@ func (mgr *VolcanojobMgr) CreatePytorchJob(c *gin.Context) {
...
 	jobName := fmt.Sprintf("py-%s", baseURL)
```
Comment (Jinghao-coding) [inline backend/internal/handler/vcjob/pytorch.go:51]: py前缀代表不了pytorch感觉 你怎么认为

File: backend/internal/handler/vcjob/tensorflow.go (hunk 3 above)
Comment (BUAA-yang-wb) [reply backend/internal/handler/vcjob/tensorflow.go:86]: 屏幕截图 2025-11-05 113125
原代码设计成了六位，可能考虑到tensflow类作业建立的数量较多。不过加上日期的话，五位应该也足够了。

File: backend/pkg/crclient/jobclient.go (hunk 1 above)
Comment (BUAA-yang-wb) [reply backend/pkg/crclient/jobclient.go:156]: 这个我删一下，和其他保持一致。

**Comment by MondayCha** [comment]
名称需要遵循 RFC 1035 规则：

> https://kubernetes.io/zh-cn/docs/concepts/overview/working-with-objects/names/#rfc-1035-label-names

RFC 1035 标签名
某些资源类型需要其名称遵循 [RFC 1035](https://tools.ietf.org/html/rfc1035) 所定义的 DNS 标签标准。也就是命名必须满足如下规则：

最多 63 个字符
只能包含小写字母、数字，以及 '-'
必须以字母开头
必须以字母数字结尾

可以对作业类型缩写一下，不重复就行，有些其他的组件（比如镜像保存）会在这个名字基础上再加后缀，容易超出63字符
